lib/rucio/tests/test_rse_expression_parser.py

```diff
@@ -84,68 +86,84 @@ def __init__(self):
     @raises(InvalidRSEExpression)
     def test_unconnected_operator(self):
         """ RSE_EXPRESSION_PARSER (CORE) Test invalid rse expression: unconnected operator"""
-        rse_expression_parser.parse_expression("TEST_RSE1|")
+        rse_expression = add_vo_to_rse_expression("TEST_RSE1|", **self.vo)
```

Comment:
These tests seem designed to test one specific aspect of the RSE expression each. Adding the VO to them all suddenly means there are multiple possible failing points (e.g. we go from 'only `|` can cause a failure' to 'failure could be any of `|&()`).

I'm wondering if there's a way to filter by VO specifically in these tests instead, e.g. by using `[... for t_rse in <result> if t_rse['vo']==self.vo['vo'] ]` on the output. Maybe not necessary in every case, but in e.g. `test_simple_rse_reference` it seems like we're no longer really testing a 'simple expression' when we tack the VO on the front. 

In some cases e.g. `test_unconnected_operator` I don't think the VO is necessary at all as it's just a syntax test.

What do you think on this?

Reply:
I see what you mean, but given that we run a full suite in both single and multi VO mode, we'd still test the simple syntax in the single vo suite (as we don't modify the vo expression if vo==def), and then test that syntax with VO additions in mutli VO mode only.
At the time I didn't really consider each test seperately, I just modified them all, so if you do think any would be better as they were I don't mind reverting.

Reply:
Good point, I forgot that in the single VO mode the expression is unchanged. The changes do clearly affect the clarity of the code, though (if it caught me out it could catch others), so I'm going to mull this one over a little bit more.

Reply:
After continued mulling, I think this is ok. The clarity of it still bothers me though, so I'm open to further discussion later if anyone has better ideas.